src/plugins/interactive_setup/server/elasticsearch_service.ts

```diff
@@ -0,0 +1,239 @@
+
+import type {
+  ElasticsearchClientConfig,
+  ElasticsearchServicePreboot,
+  ICustomClusterClient,
+  Logger,
+  ScopeableRequest,
+} from 'src/core/server';
+
+import { ElasticsearchConnectionStatus } from '../common';
+import { getDetailedErrorMessage } from './errors';
+
+interface EnrollParameters {
+  apiKey: string;
+  hosts: string[];
+  // TODO: Integrate fingerprint check as soon core supports this new option:
+  // https://github.com/elastic/kibana/pull/108514
+  caFingerprint?: string;
+}
+
+export interface ElasticsearchServiceSetupDeps {
+  /**
+   * Core Elasticsearch service preboot contract;
+   */
+  elasticsearch: ElasticsearchServicePreboot;
+
+  /**
+   * Interval for the Elasticsearch connection check (whether it's configured or not).
+   */
+  connectionCheckInterval: Duration;
+}
+
+export interface ElasticsearchServiceSetup {
+  /**
+   * Observable that yields the last result of the Elasticsearch connection status check.
+   */
+  connectionStatus$: Observable<ElasticsearchConnectionStatus>;
+
+  /**
+   * Iterates through provided {@param hosts} one by one trying to call Kibana enrollment API using
+   * the specified {@param apiKey}.
+   * @param apiKey The ApiKey to use to authenticate Kibana enrollment request.
+   * @param hosts The list of Elasticsearch node addresses to enroll with. The addresses are supposed
+   * to point to exactly same Elasticsearch node, potentially available via different network interfaces.
+   */
+  enroll: (params: EnrollParameters) => Promise<EnrollResult>;
+}
+
+/**
+ * Result of the enrollment request.
+ */
+export interface EnrollResult {
+  /**
+   * Host address of the Elasticsearch node that successfully processed enrollment request.
+   */
+  host: string;
+  /**
+   * PEM CA certificate for the Elasticsearch HTTP certificates.
+   */
+  ca: string;
+  /**
+   * Service account token for the "elastic/kibana" service account.
+   */
+  serviceAccountToken: { name: string; value: string };
+}
+
+export class ElasticsearchService {
+  /**
+   * Elasticsearch client used to check Elasticsearch connection status.
+   */
+  private connectionStatusClient?: ICustomClusterClient;
+  constructor(private readonly logger: Logger) {}
+
+  public setup({
+    elasticsearch,
+    connectionCheckInterval,
+  }: ElasticsearchServiceSetupDeps): ElasticsearchServiceSetup {
+    const connectionStatusClient = (this.connectionStatusClient = elasticsearch.createClient(
+      'ping'
+    ));
+
+    return {
+      connectionStatus$: timer(0, connectionCheckInterval.asMilliseconds()).pipe(
+        exhaustMap(() => {
+          return from(connectionStatusClient.asInternalUser.ping()).pipe(
+            map(() => ElasticsearchConnectionStatus.Configured),
+            catchError((pingError) =>
+              of(
+                pingError instanceof errors.ConnectionError
+                  ? ElasticsearchConnectionStatus.NotConfigured
+                  : ElasticsearchConnectionStatus.Configured
```

Comment:
__note:__ I am a bit cautious here and don't try to handle other potentially acceptable errors (e.g. Timeout), it's better to be 100% sure here that Kibana is really unable to connect to ES.